creator-node/src/routes/nodeSync.js

```diff
@@ -139,14 +203,16 @@ module.exports = function (app) {
 
     // Get & return latestBlockNumber for wallet
     const cnodeUser = await models.CNodeUser.findOne({ where: { walletPublicKey } })
-    const latestBlockNumber = cnodeUser ? cnodeUser.latestBlockNumber : -1
+    const latestBlockNumber = (cnodeUser) ? cnodeUser.latestBlockNumber : -1
+    const clockValue = (cnodeUser) ? cnodeUser.clock : -1
 
-    return successResponse({ walletPublicKey, latestBlockNumber })
+    return successResponse({ walletPublicKey, latestBlockNumber, clockValue })
   }))
 }
 
-async function _nodesync (req, walletPublicKeys, creatorNodeEndpoint) {
+async function _nodesync (req, walletPublicKeys, creatorNodeEndpoint, dbOnlySync) {
```

Comment:
changes in this function
- new dbOnlySync param which if true doesn't peer with bootstrap nodes, doesn't build replica set for user, forces a sync on the secondary regardless of blocknumber check, ignores file system writes for files
- adds clockrecord db entry save
- returns an errorObj if there was an error. does not throw because our current sync doesn't handle errors. the new vector clock sync handles the error here https://github.com/AudiusProject/audius-protocol/pull/772/files#diff-6296aeaad2e77b78d513911233301094R189

Reply:
👍